src/components/postCompontents/EditPostPage.jsx

```diff
@@ -0,0 +1,73 @@
+import { useParams } from "react-router-dom";
+import { getRequest, putRequest } from "../../utilites/apiRequests";
+import { useEffect, useState } from "react";
+
+export const EditPostPage = () => {
+	const [post, setPost] = useState(null);
+	const [formData, setFormData] = useState(null);
+	const [successfullSubmit, setSuccesfullSubmit] = useState(false);
+	const { postId } = useParams();
+
+	useEffect(() => {
+		getRequest(`/post/${postId}`).then((response) => {
+			setPost({ ...response });
+			setFormData({ ...response });
+		});
+	}, [postId]);
+
+	const handleInput = (event) => {
+		const { name, value } = event.target;
+		setFormData({ ...formData, [name]: value });
+		setSuccesfullSubmit(false);
+	};
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		putRequest(`/post/${postId}`, formData).then(setSuccesfullSubmit(true));
```

Comment:
You're setting setSuccesfullSubmit(true) immediately after calling putRequest. Since putRequest is asynchronous, you might want to wait until you know the request was successful before indicating to the user that their changes have been saved